tools/plaso_pubsub.py

```diff
@@ -0,0 +1,325 @@
+    form_data = {u'username': username, u'password': password}
+    session.post(u'{0:s}/login/'.format(self.host_url), data=form_data)
+    return session
+
+  def CreateSketch(self, name, description):
+    resource_url = u'{0:s}/api/v1/sketches/'.format(self.host_url)
+    form_data = {u'name': name, u'description': description}
+    response = self.session.post(resource_url, json=form_data)
+    response_dict = response.json()
+    sketch_id = response_dict[u'objects'][0]['id']
+    return sketch_id
+
+  def UploadTimeline(self, timeline_name, plaso_storage_path):
+    resource_url = u'{0:s}/api/v1/upload/'.format(self.host_url)
+    files = {'file': open(plaso_storage_path, 'rb')}
+    form_data = {u'name': timeline_name}
+    response = self.session.post(resource_url, files=files, data=form_data)
+    response_dict = response.json()
+    index_id = response_dict[u'objects'][0]['id']
+    return index_id
+
+  def AddTimelineToSketch(self, sketch_id, index_id):
+    resource_url = u'{0:s}/api/v1/sketches/{0:d}/'.format(self.host_url,
+                                                          sketch_id)
+    form_data = {u'timelines': [index_id]}
+    response = self.session.post(resource_url, json=form_data)
+    response_dict = response.json()
+    timeline_id = response_dict[u'objects'][0]['id']
+    return timeline_id
+
+
+class PlasoProcessor(object):
+
+  def __init__(self):
+    config.LoadConfig()
+
+  def _WaitForOperation(self, client, operation):
+    """Wait for an operation to complete."""
+    WriteToStdOut(u'Waiting for operation to complete')
+    while True:
+      result = client.zoneOperations().get(project=config.PROJECT,
+                                           zone=config.ZONE,
+                                           operation=operation).execute()
+      WriteToStdOut(u'Status: {0:s}'.format(result[u'status']))
+      if result[u'status'] == u'DONE':
+        if u'error' in result:
+          WriteToStdErr(result)
+          raise RuntimeError(result[u'error'][u'errors'][0][u'message'])
+        return result
+      time.sleep(1)
+
+  def _AttachDisk(self, client, disk):
+    """Attaches a persistent disk to the machine."""
+    WriteToStdOut(u'Attaching disk')
+    operation = client.instances().attachDisk(
+        instance=config.INSTANCE,
+        project=config.PROJECT,
+        zone=config.ZONE,
+        body={u'deviceName': config.DEVICE_NAME,
+              u'source': disk}).execute()
+    self._WaitForOperation(client, operation[u'name'])
+
+  def _DetachDisk(self, client):
+    """Detaches the disk from the machine."""
+    WriteToStdOut(u'Detaching disk')
+    operation = client.instances().detachDisk(
+        instance=config.INSTANCE,
+        project=config.PROJECT,
+        zone=config.ZONE,
+        deviceName=config.DEVICE_NAME).execute()
+    self._WaitForOperation(client, operation[u'name'])
+
+  def _CopyFileToBucket(self, client, filename, object_name):
```

Comment:
I think we should move the GCE stuff out to its own class. But we can do that in a later CL. What do you think?